### ISSUE TITLE
building functional tests failing with openjdk-tests/functional/build.xml:52: java.lang.NullPointerException

### ISSUE THREAD

**Issue by pshipton** [eclipse/openj9#3319]
First started occurring in omr acceptance builds
https://ci.eclipse.org/openj9/view/Pipelines/job/Pipeline-OMR-Acceptance/851/
(note some previous builds were cancelled)
OpenJ9: 3cc0d66
OMR: 4e4fa19
OpenJDK8: 1ba8f1a
OpenJDK11: 4f9e2ba
OpenJDK9: 364c38a
Only seems to occur on Java 11 linux_x86

failure 1 https://ci.eclipse.org/openj9/job/Test-sanity.functional-JDK11-linux_x86-64_cmprssptrs/93/console
ub16-x86-1
failure 2 https://ci.eclipse.org/openj9/job/Test-sanity.functional-JDK11-linux_x86-64_cmprssptrs/95/console
ub16-x86-6
failure 3 https://ci.eclipse.org/openj9/job/Test-sanity.functional-JDK11-linux_x86-64_cmprssptrs/96/console
ub16-x86-5
failure 4 https://ci.eclipse.org/openj9/job/Test-sanity.functional-JDK11-linux_x86-64/91/console
ub16-x86-1

Occurred in the OpenJ9 nightly builds

https://ci.eclipse.org/openj9/job/Test-extended.functional-JDK11-linux_x86-64_cmprssptrs/55/console
ub16-x86-2
https://ci.eclipse.org/openj9/job/Test-sanity.functional-JDK11-linux_x86-64_cmprssptrs/100/console
ub16-x86-2
https://ci.eclipse.org/openj9/job/Test-sanity.functional-JDK11-linux_x86-64/95/console
ub16-x86-1

```
14:51:28 BUILD FAILED
14:51:28 /home/jenkins/workspace/Test-sanity.functional-JDK11-linux_x86-64_cmprssptrs/openjdk-tests/TestConfig/scripts/build_test.xml:71: The following error occurred while executing this line:
14:51:28 /home/jenkins/workspace/Test-sanity.functional-JDK11-linux_x86-64_cmprssptrs/openjdk-tests/functional/build.xml:52: java.lang.NullPointerException
14:51:28 	at org.apache.tools.ant.ComponentHelper.initSubProject(ComponentHelper.java:235)
14:51:28 	at org.apache.tools.ant.Project.initSubProject(Project.java:290)
14:51:28 	at org.apache.tools.ant.Project.createSubProject(Project.java:280)
14:51:28 	at org.apache.tools.ant.taskdefs.Ant.init(Ant.java:168)
14:51:28 	at org.apache.tools.ant.taskdefs.SubAnt.createAntTask(SubAnt.java:577)
14:51:28 	at org.apache.tools.ant.taskdefs.SubAnt.execute(SubAnt.java:296)
14:51:28 	at org.apache.tools.ant.taskdefs.SubAnt.execute(SubAnt.java:224)
14:51:28 	at org.apache.tools.ant.UnknownElement.execute(UnknownElement.java:293)
14:51:28 	at jdk.internal.reflect.GeneratedMethodAccessor6.invoke(Unknown Source)
14:51:28 	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
14:51:28 	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
14:51:28 	at org.apache.tools.ant.dispatch.DispatchUtils.execute(DispatchUtils.java:106)
14:51:28 	at org.apache.tools.ant.Task.perform(Task.java:348)
14:51:28 	at org.apache.tools.ant.taskdefs.Sequential.execute(Sequential.java:68)
14:51:28 	at net.sf.antcontrib.logic.IfTask.execute(IfTask.java:217)
14:51:28 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
14:51:28 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
14:51:28 	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
14:51:28 	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
14:51:28 	at org.apache.tools.ant.dispatch.DispatchUtils.execute(DispatchUtils.java:106)
14:51:28 	at org.apache.tools.ant.TaskAdapter.execute(TaskAdapter.java:155)
14:51:28 	at org.apache.tools.ant.UnknownElement.execute(UnknownElement.java:293)
14:51:28 	at jdk.internal.reflect.GeneratedMethodAccessor6.invoke(Unknown Source)
14:51:28 	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
14:51:28 	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
14:51:28 	at org.apache.tools.ant.dispatch.DispatchUtils.execute(DispatchUtils.java:106)
14:51:28 	at org.apache.tools.ant.Task.perform(Task.java:348)
14:51:28 	at org.apache.tools.ant.Target.execute(Target.java:435)
14:51:28 	at org.apache.tools.ant.Target.performTasks(Target.java:456)
14:51:28 	at org.apache.tools.ant.Project.executeSortedTargets(Project.java:1405)
14:51:28 	at org.apache.tools.ant.helper.SingleCheckExecutor.executeTargets(SingleCheckExecutor.java:38)
14:51:28 	at org.apache.tools.ant.Project.executeTargets(Project.java:1260)
14:51:28 	at org.apache.tools.ant.taskdefs.Ant.execute(Ant.java:441)
14:51:28 	at org.apache.tools.ant.taskdefs.SubAnt.execute(SubAnt.java:309)
14:51:28 	at org.apache.tools.ant.taskdefs.SubAnt.execute(SubAnt.java:224)
14:51:28 	at org.apache.tools.ant.UnknownElement.execute(UnknownElement.java:293)
14:51:28 	at jdk.internal.reflect.GeneratedMethodAccessor6.invoke(Unknown Source)
14:51:28 	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
14:51:28 	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
14:51:28 	at org.apache.tools.ant.dispatch.DispatchUtils.execute(DispatchUtils.java:106)
14:51:28 	at org.apache.tools.ant.Task.perform(Task.java:348)
14:51:28 	at org.apache.tools.ant.Target.execute(Target.java:435)
14:51:28 	at org.apache.tools.ant.Target.performTasks(Target.java:456)
14:51:28 	at org.apache.tools.ant.Project.executeSortedTargets(Project.java:1405)
14:51:28 	at org.apache.tools.ant.Project.executeTarget(Project.java:1376)
14:51:28 	at org.apache.tools.ant.helper.DefaultExecutor.executeTargets(DefaultExecutor.java:41)
14:51:28 	at org.apache.tools.ant.Project.executeTargets(Project.java:1260)
14:51:28 	at org.apache.tools.ant.Main.runBuild(Main.java:853)
14:51:28 	at org.apache.tools.ant.Main.startAnt(Main.java:235)
14:51:28 	at org.apache.tools.ant.launch.Launcher.run(Launcher.java:285)
14:51:28 	at org.apache.tools.ant.launch.Launcher.main(Launcher.java:112)
...
03:40:13 makefile:82: recipe for target 'compile' failed
```

**Comment by pshipton** [comment]
An omr promotion at openj9 sha 34b9d4f promoted. Looking at the changes between 34b9d4f and 3cc0d66 it seems likely that https://github.com/eclipse/openj9/pull/3295 , which has the affect of enabling shared classes by default on xlinux, is causing the problem.

**Comment by pshipton** [comment]
Shared classes was initially enabled by default in #2449 but the changes did not affect xlinux.

**Comment by pshipton** [comment]
FWIW I ran 6+ grinders each on the latest Java 11 xlinux compressedrefs/largeheap to build the functional sanity testing without issue. Trying the Build-Test-All which runs nightly, customized to run jdk11 xlinux* functional.sanity.

**Comment by pshipton** [comment]
I'm calling this fixed by #3320. I'm also going to revert #2449 to disable shared classes by default for all platforms.
